plugins/aws/cloudwatchlogs/logRetentionPeriod.js

```diff
@@ -0,0 +1,66 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'CloudWatch Log Retention Period',
+    category: 'CloudWatchLogs',
+    description: 'Ensures that the CloudWatch Log retention period is set above a specified length of time.',
+    more_info: 'Retention settings can be used to specify how long log events are kept in CloudWatch Logs. Expired log events get deleted automatically.',
+    recommended_action: 'Ensure CloudWatch logs are retained for at least 90 days.',
+    link: 'https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Working-with-log-groups-and-streams.html',
+    apis: ['CloudWatchLogs:describeLogGroups'],
+    settings: {
+        minimum_log_retention_period: {
+            name: 'CloudWatch Log Minimum Retention Period',
+            description: 'If set, CloudWatch Logs log groups should have a retention setting greater or equal to this value',
+            regex: '^[0-9]*$',
+            default: 0
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var config = {
+            minimum_log_retention_period: settings.minimum_log_retention_period || this.settings.minimum_log_retention_period.default
+        };
+
+        if (config.minimum_log_retention_period === false) return callback();
```

Comment:
if the default is 0 wouldn't this callback? the plugin will never run. I don't think we should make this an opt-in plugin that has to be turned on. I think its a good plugin and should be on by default

Reply:
Yes, that was the idea. But you are right. I've changed the default value to be 90